TOY-37 특정 시간대에 글작성을 권유하는 알림 기능 구현

pby2017 opened the pull request:
코드 리뷰 부탁드립니다.
WorkManager로 구현하였습니다.

특정 시간에 알람이 오는 것을 테스트해보시려면,
`ReminderWork.kt` 파일의
`enqueueReminder()` 메소드의 아래 내용을 바꿔주시고 실행하셔서
설정 화면에서 알람 설정 토글 (On/Off)을 On으로 바꿔주시면 됩니다.

```kotlin
set(Calendar.HOUR_OF_DAY, 12)
set(Calendar.MINUTE, 0)
set(Calendar.SECOND, 0)
set(Calendar.MILLISECOND, 0)
```
```kotlin
// set(Calendar.HOUR_OF_DAY, 12)
// set(Calendar.MINUTE, 0)
// set(Calendar.SECOND, 0)
add(Calendar.SECOND, 2) // 2 자리에 (0~59)초 단위 숫자 입력
set(Calendar.MILLISECOND, 0)
```

이후 티켓 생성해서 
1. 알람의 다른 작업으로 이동하는 버튼 눌러서 이동했을 때 알람이 종료되지 않는 문제 해결
구현하려고 합니다.

----------

#### [2019/06/27 추가]
멀티덱스 설정 코드를 추가했습니다.

pby2017 committed:
TOY-37 낮 12시 알림 생성 구현

pby2017 committed:
TOY-37 알람 스케줄 시작 코드를 GlobalApplication 으로 이동

Merge branch 'develop' into TOY-37-diary-notification

pby2017 committed:
TOY-37 충돌 해결

Merge branch 'develop' into TOY-37-diary-notification

# Conflicts:
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/GlobalApplication.kt
#	app/src/main/res/values/strings.xml

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/notification/JobSchedulerStart.java:
```diff
@@ -0,0 +1,37 @@
+    private static final int JOB_ID = 1111;
+
+    public static void start(Context context) {
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.LOLLIPOP) {
```
import android.os.Build 로 변경해주세요.

pby2017 replied:
네,
숙지하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/notification/MyBroadcastReceiver.java:
```diff
@@ -0,0 +1,95 @@
+package kr.co.yogiyo.rookiephotoapp.notification;
```
몇 줄 안되는 코드는 코틀린으로 작성해보세요~

pby2017 replied:
네,
반영하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/notification/NotificationConstant.java:
```diff
@@ -0,0 +1,17 @@
+package kr.co.yogiyo.rookiephotoapp.notification;
```
여기도 그냥 코틀린으로 작성해봐요~

pby2017 replied:
네,
반영하겠습니다.

bonchangkoo reviewed (commented):
Activity나 Fragment 같이 화면이나 복잡한 로직이 없으면 처음부터 코틀린으로 작성해보세요.

pby2017 committed:
TOY-37 WorkManager 사용

pby2017 commented:
> Activity나 Fragment 같이 화면이나 복잡한 로직이 없으면 처음부터 코틀린으로 작성해보세요.

네,
반영하겠습니다.

pby2017 committed:
TOY-37 충돌 해결

Merge branch 'develop' into TOY-37-diary-notification

# Conflicts:
#	app/build.gradle
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/Constants.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/GlobalApplication.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/settings/SettingsFragment.java

pby2017 committed:
TOY-37 알림 기능 구현 이후 멀티덱스 옵션 추가

hyejin830 commented:
빌드 오류가 발생합니다.
clean project 후에 make project 시도해보았고, 껏다가 켜봤는데도 계속 동일한 오류가 발생하네요...
확인 부탁드립니다!

아래는 에러 메세지입니다
```
Class 'ReminderWork' is not abstract and does not implement abstract base class member public abstract fun isDisposed(): Boolean defined in io.reactivex.Scheduler.Worker 

Too many arguments for public constructor Worker() defined in io.reactivex.Scheduler.Worker 
 
One type argument expected for class Result

Type inference failed. Expected type mismatch: inferred type is Class but Class was expected  

```

hyejin830 commented:
알람 아이콘 적용해주세요~! 확인 부탁드립니다

hyejin830 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -21,7 +21,8 @@
             android:theme="@style/Theme.AppCompat.Light.NoActionBar" />
-        <activity android:name=".MainActivity">
+        <activity android:name=".MainActivity"
+            android:noHistory="true">
```
[질문]이 브랜치에서 noHistory는 어떡 역할인거에요?!

pby2017 replied:
MainActivity에서 다른 액티비티 시작할 때
기본적으로 MainActivity를 스택에 넣는데
`noHistory="true` 를 추가하면 스택에 안넣습니다

CameraActivity 에서 뒤로가기를 누르면
MainActivity가 보이지 않고 바로 앱이 종료됩니다.

----------

TOY-24 카메라 브랜치에도 추가되어있고
이 브랜치에서는 필요 없는 코드여서 제거하겠습니다.

hyejin830 commented:
확인 완료하였습니다

pby2017 commented:
> 알람 아이콘 적용해주세요~! 확인 부탁드립니다

@hyejin830 님,
피드백 반영하겠습니다.

[링크](https://kr.piliapp.com/emoji/list/thinking/) 에 있는 이모티콘 복사해서 그대로 붙여넣겠습니다.

pby2017 committed:
TOY-37 피드백 반영

- androidManifest 파일의 noHistory 옵션 제거
- 아이콘 크기 변경
- 알림 텍스트에 이모티콘 추가

bonchangkoo reviewed (approved):
확인했어요~

pby2017 committed:
TOY-37 충돌 해결
